### PR TITLE
Implement post event participant attendance

### LINKED ISSUE #1220: 🤸‍♀️ Post-event participant attendance reporting
## Description
A user should be able to report which participants attended the event, and which ones did not show up. This will help searching for participants in the future.

We call this "confirmed attendance", as opposed to "booked participants", to indicate that while "confirmed attendance" means some human confirmed that a participant attended, the lack of "confirmed attendance" does not mean the person definitely did not attend.

Also see epic #1214 for context.

## Relevant Job Stories
* When certain events have taken place, I want to feedback into the system who actually showed up, so that I can keep track of attendance and search accordingly
* When a large, open-meeting type event has taken place with a very large number of attendees, I want whatever data is available (e.g. sign-ups) to be usable without any further action, so that I don’t have to log the attendance of every individual for it to be usable

## Prerequisites
Don't start this until #1219 has been merged.

EDIT: See comment below for API details.

## Requirements
- [x] After an event has passed, the summary at the top of the participants tab updates to replace "Notifications" with "Confirmed attendance"
  - [x] The numbers under "Confirmed attendance" shows confirmed attended against booked participants
  - [x] If there are any "No-shows", that number is also displayed
  - [x] If there are any participants whose attendance has not been recorded, show "Record attendance" button, that when clicked scrolls down to the "Booked participants" list
- [x] In the "booked participants" list, the action button (previously "Cancel") is replaced with three buttons:
  - [x] Cancelled, which when clicked moves the person to the "Cancelled" section (from #1219)
  - [x] No-show, which when clicked records the participant as "no-show" in the API
  - [x] Attended, which when clicked records the participant as "attended" in the API
  - [x] Once attendance has been recorded, the buttons are replaced with a drop-down with the same three options

## Possible implementations
The API for this has not yet been implemented.

### Summary after event
image

### Summary after recording attendance
image

### List after event
image

## Open questions
What will the API for this look like? See thread below for updates on when the API has been implemented.

### PR DESCRIPTION
**Pull request by niklasva82**
## Description
This PR adds the ability to set wether participants attended after the event.

## Screenshots
bild

## Changes
* Adds buttons for attended and noshow
* Refactor the way buttons and dropdowns were rendered into separate components, it was getting a bit too messy.

## Related issues
Resolves #1220

### CODE REVIEW

**Review by richardolsson — CHANGES_REQUESTED** [review]
I think the solution looks great in general, but it's missing a couple of features.

* The "book" button under signups as been lost. I'm not sure what would be the best solution here, whether there should be attendance buttons in the signups list after the event or not, but at the very least we still need the book button, especially before an event takes place.
* The requirements concerning the summary at the top of the page are not met. Please have a look at the requirements list and design mockups in the issue #1220 for details.

File: src/features/events/components/ParticipantListSection.tsx [hunk 1]
```diff
@@ -164,47 +241,90 @@ const ParticipantListSection: FC<ParticipantListSectionListProps> = ({
...
-                onClick={noPropagate(() => {
-                  model.addParticipant(params.row.id);
-                })}
-                variant="outlined"
-              >
-                {messages.eventParticipantsList.buttonBook()}
-              </Button>
-            </>
+            <Buttons
+              options={[
```
[the remark this answers is not in the document]
Comment (niklasva82) [reply src/features/events/components/ParticipantListSection.tsx:245]: It seems I forgot about the book button when reimplementing. This has been fixed.

The options are different before and after for the booked list, but not the others.

**Comment by niklasva82** [comment]
[Screencast from 2023-05-10 00-56-09.webm](https://github.com/zetkin/app.zetkin.org/assets/14962107/89b6d6de-c824-447c-849a-4cd1875a3f73)

File: src/features/events/components/ParticipantSummaryCard.tsx [hunk 2]
```diff
@@ -145,44 +147,64 @@ const ParticipantSummaryCard: FC<ParticipantSummaryCardProps> = ({ model }) => {
...
-              {remindedParticipants < availParticipants && (
-                <Tooltip
-                  arrow
-                  placement="top-start"
-                  title={
-                    contactPerson == null
-                      ? messages.participantSummaryCard.remindButtondisabledTooltip()
-                      : ''
-                  }
+          {new Date(eventData.start_time) > new Date() ? (
```
Comment (richardolsson) [inline src/features/events/components/ParticipantSummaryCard.tsx:150]: I find it very difficult to read these very large ternary expressions in the middle of JSX, and would appreciate if you can shorten it in one of several ways:

### 1. Separate components
Convert the contents to separate components (defined in the same module) and use as such:

```tsx
{new Date(eventData.start_time) > new Date() ? <ReminderInfo …/> : <AttendanceInfo …/>}
```

### 2. Functions
Move the contents to functions (defined inside of this component) and use as such:

```tsx
{new Date(eventData.start_time) > new Date() ? renderReminderInfo() : renderAttendanceInfo()}
```

File: src/features/events/components/ParticipantSummaryCard.tsx [hunk 3]
```diff
@@ -145,44 +147,64 @@ const ParticipantSummaryCard: FC<ParticipantSummaryCardProps> = ({ model }) => {
...
+            <Box display="flex" flexDirection="column">
+              <Typography color={'secondary'}>
+                {messages.participantSummaryCard.confirmed()}
+              </Typography>
+              <Box alignItems="flex-end" display="flex">
+                <Typography variant="h4">{`${confirmedParticipants}/${availParticipants}`}</Typography>
+                <Typography
+                  color={'GrayText'}
+                  sx={{ fontSize: '1.7em', marginLeft: '0.3em' }}
+                  variant="h4"
```
Comment (richardolsson) [inline src/features/events/components/ParticipantSummaryCard.tsx:199]: This should only be visible when `noshowParticipants > 0`, as stated in the requirements in the issue.

**Review by richardolsson — CHANGES_REQUESTED** [review]
We're getting closer but I still found one bug, and one code style issue that is related to my personal taste but I think is arguably better for readibility.

File: src/features/events/components/ParticipantSummaryCard.tsx (hunk 2 above)
Comment (richardolsson) [reply src/features/events/components/ParticipantSummaryCard.tsx:157]: I started working on this but decided to leave it as I think this component could do with a bigger refactor than what I have time to do right now.

**Review by richardolsson — APPROVED** [review]
Ready to merge! 🎉

**Pull request merged**